Sources/Utility/Process.swift

```diff
@@ -165,6 +178,35 @@ public final class Process: ObjectIdentifierProtocol {
         self.redirectOutput = redirectOutput 
     }
 
+    /// Returns true if the given program is present and executable in search path.
+    ///
+    /// The program can be executable name, relative path or absolute path.
+    func findExecutable(_ program: String) -> Bool {
+        return Process.executablesQueue.sync {
+            // Check if we already have a value for the program.
+            if let value = Process.validatedExecutablesMap[program] {
+                return value
+            }
+            // FIXME: This can be cached.
+            let envSearchPaths = Utility.getEnvSearchPaths(
+                pathString: getenv("PATH"),
+                currentWorkingDirectory: currentWorkingDirectory
+            )
+            // Lookup the executable.
+            let value = Utility.lookupExecutablePath(
+                filename: program, searchPaths: envSearchPaths) != nil
+            Process.validatedExecutablesMap[program] = value
+            return value
+        }
+    }
+
+    /// Validate that the program provided exists on the search path and is executable.
+    func validateArgumentIsExecutable(_ program: String) throws {
```

Comment:
We should move the guard to launch() and just write test for `findExecutable()` and `launch()`